voucher-mission/src/main/java/org/prgrms/vouchermission/MemoryVoucherRepository.java

```diff
@@ -0,0 +1,25 @@
+package org.prgrms.vouchermission;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+@Repository
+public class MemoryVoucherRepository implements VoucherRepository {
+
+    private final Map<UUID, Voucher> storage = new ConcurrentHashMap<>();
+
+    @Override
+    public Voucher insert(Voucher voucher) {
+        storage.put(voucher.getVoucherId(), voucher);
+        return voucher;
+    }
+
+    @Override
+    public List<Voucher> findAll() {
+        return storage.values()
+                .stream().collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    @Override
    public List<Voucher> findAll() {
           return storage.values().toList();
    }
```
이렇게는 불가능할까요? 또 기존 코드와 어떤 차이점이 있을까용

Reply:
후자로 바꾸게 될 경우, 값 변경이 불가능한 List가 반환된다는 차이가 있습니다.
[refactor : 바우처 생성 검증로직 추가 및 findAll() 리턴값 변경](https://github.com/prgrms-be-devcourse/springboot-basic/pull/690/commits/73846fe271ab1a375053c08d03ad51133e8b434d)